pkg/sql/catalog/bootstrap/metadata.go

```diff
@@ -452,6 +453,75 @@ func addSystemDatabaseToSchema(
 	addSystemDescriptorsToSchema(target)
 	addSplitIDs(target)
 	addZoneConfigKVsToSchema(target, defaultZoneConfig, defaultSystemZoneConfig)
+	addSystemTenantEntry(target)
+}
+
+// addSystemTenantEntry adds a kv pair to system.tenants to define the initial
+// system tenant entry.
+func addSystemTenantEntry(target *MetadataSchema) {
+	info := descpb.TenantInfo{
+		ID:    roachpb.SystemTenantID.ToUint64(),
+		Name:  "system",
+		State: descpb.TenantInfo_ACTIVE,
+	}
+	infoBytes, err := protoutil.Marshal(&info)
+	if err != nil {
+		panic(err)
+	}
+
+	// Find the system.tenant descriptor in the newly created catalog.
+	// TODO(knz): Is there a better way?
+	var tableDesc catalog.TableDescriptor
+	for _, desc := range target.descs {
+		if desc.GetName() == "tenants" {
+			tableDesc = desc.(catalog.TableDescriptor)
+			break
+		}
+	}
+	if tableDesc == nil {
+		// No system.tenant table (we're likely in a secondary
+		// tenant). Nothing to do.
+		return
+	}
+
+	// TODO(knz): What if more indexes or columns are added to the table
+	// after this code has been written? I sure wish I could express
+	// this using SQL statements (perhaps as a separate code generator
+	// that would provide raw bytes that I can paste here?)
```

Comment:
I think your code already deals adequately with adding more indexes. As for more columns, you could probably add an if-check on `len(tableDesc.PublicColumns()) == len(values)` and panic, which would act as a poor person's assertion. Perhaps do so in an `init` function. 

With such an assertion, constructing `colIDtoRowIndex` with
```
for i, c := range tableDesc.PublicColumns() {
   colIDtoRowIndex.Set(c.GetID(), i)
}
```
should be acceptable.